### ISSUE TITLE
saveMany doesn't work on new models in PHP 5.6.12

### ISSUE THREAD

**Issue by zulrang** [yajra/laravel-oci8#91]
The following code will cause a database error stating that "integrity constraint violated - parent key not found":

```php
$question = new QuizQuestion();
$question->text = "This is a question";
$question->save();

$responses = [
	new QuizResponse(['test' => 'Answer 1']),
	new QuizResponse(['text'=> 'Answer 2']),
	new QuizResponse(['text' => 'Answer 3', 'is_correct' => 'Y'])
];

$question->responses()->saveMany($responses);
```

That will produce a query that attempts to set the question_id to zero or null, even though stepping through the code shows that it's binding the correct parent ID before the execution of the query.

It seems there's some type of memory mapping issue with PHP references and binding, I believe the issue is the following line of code in OracleProcessor.php:146 within the function bindValuesAndReturnCounter:

```php
$this->statement->bindValue($counter, ($value), $param);
```

The use of parentheses around the $value seems to be taking advantage of a bug in PHP:  https://bugs.php.net/bug.php?id=55222

The code above works fine in PHP 5.6.4, but no longer works in 5.6.12.  Somewhere between there the reference counting bug was either addressed or regressed.  I'm going to look into the issue further and report back.

**Comment by zulrang** [comment]
So far, I've found that it works correctly if ```$question->id``` is a string, but not if it is an integer.  The database schema has the column defined as a NUMBER.

**Comment by yajra** [comment]
@zulrang, I just tested `saveMany` on my homestead with `PHP 5.6.11-1+deb.sury.org~utopic+1 (cli)` and seems to work well.  Will try to setup 5.6.12 on my free time. Thanks for digging this up. I'm a bit occupied atm and may not be able to check this soon.

**Comment by zulrang** [comment]
I'm going to do more testing tomorrow.  I'm thinking it has something to do with the supplied length in oci_bind_by_name, as I vaguely recall an issue with that in other libraries.

**Comment by zulrang** [comment]
On a side note, what IS with the extra parens around $value in the code?
```php
$this->statement->bindValue($counter, ($value), $param);
```

I've never seen this before.

**Comment by yajra** [comment]
hmm. Thats weird and must be a typo. It has no special meaning at all. I tried digging the commit history and it was there since I implemented `insertGetId` method last 2/7/2014. ^_^

**Comment by zulrang** [comment]
Okay, good to know I'm not crazy and don't have to chase my tail. ^_^

**Comment by zulrang** [comment]
I discovered the issue, and traced the problem back to a pull request (https://github.com/yajra/laravel-oci8/pull/60) I submitted earlier in the year.  It's not an issue between PHP version, it's an issue between 32- and 64-bit. The fix needs to be applied to https://github.com/yajra/pdo-via-oci8.  

pdo-via-oci8 doesn't account for the PDO::PARAM_INPUT_OUTPUT flag.  Adding the following line to Pdo\Oci8\Statement.php fixes the issue:

```php
// strip INOUT type for oci
$dataType &= ~PDO::PARAM_INPUT_OUTPUT;
```

I'll create a patch and pull request over there.
